src/app/GitUI/LeftPanel/FavoritesTree.cs

```diff
@@ -0,0 +1,168 @@
+            }
+
+            Refresh(new FilteredGitRefsProvider(UICommands.Module).GetRefs, true);
+        }
+
+        protected override Nodes FillTree(IReadOnlyList<IGitRef> branches, CancellationToken token)
+        {
+            FavoriteNode remoteNode = new(this, TranslatedStrings.Remotes, nameof(Images.BranchRemoteRoot)) { Visible = true };
+            FavoriteNode localNode = new(this, TranslatedStrings.Local, nameof(Images.BranchLocalRoot)) { Visible = true };
+
+            Dictionary<string, BaseRevisionNode> pathToNodesRemote = [];
+            Dictionary<string, BaseRevisionNode> pathToNodeLocal = [];
+            IDictionary<string, AheadBehindData> aheadBehindDataLocal = _aheadBehindDataProvider?.GetData();
+
+            foreach (IGitRef branch in branches)
+            {
+                token.ThrowIfCancellationRequested();
+
+                if (branch.IsRemote)
+                {
+                    CreateRemoteBranchTree(branch, pathToNodesRemote, remoteNode.Nodes, aheadBehindDataLocal);
+                }
+                else if (branch.IsHead)
+                {
+                    CreateLocalBranchTree(branch, pathToNodeLocal, localNode.Nodes, aheadBehindDataLocal);
+                }
+            }
+
+            _cachedFavorites.Favorites.ToList().Except(branches.Select(p => p.ObjectId?.ToString())).ForEach(p => _cachedFavorites.Remove(p));
+
+            Nodes nodes = new(this);
+            nodes.AddNode(localNode);
+            nodes.AddNode(remoteNode);
+
+            return nodes;
+        }
+
+        protected override void PostFillTreeViewNode(bool firstTime)
+        {
+            if (firstTime)
+            {
+                TreeViewNode.Collapse();
+            }
+        }
+
+        private void CreateLocalBranchTree(IGitRef branch, Dictionary<string, BaseRevisionNode> pathToNodes, Nodes nodes, IDictionary<string, AheadBehindData>? aheadBehindData)
+        {
+            Validates.NotNull(branch.ObjectId);
+
+            if (!_cachedFavorites.Contains(branch.ObjectId.ToString()))
+            {
+                return;
+            }
+
+            string currentBranch = _revisionGridInfo.GetCurrentBranch();
+
+            LocalBranchNode branchNode = new(this, branch.ObjectId, branch.Name, branch.Name == currentBranch, true);
+
+            if (aheadBehindData?.TryGetValue(branchNode.FullPath, out AheadBehindData aheadBehind) is true)
+            {
+                branchNode.UpdateAheadBehind(aheadBehind.ToDisplay(), aheadBehind.RemoteRef);
+            }
+
+            BaseRevisionNode parent = branchNode.CreateRootNode(pathToNodes, CreatePathNode);
+
+            if (parent is not null)
+            {
+                nodes.AddNode(parent);
+            }
+
+            BaseRevisionNode CreatePathNode(Tree tree, string parentPath)
+            {
+                BranchPathNode branchPathNode = new(tree, parentPath);
+
+                return branchPathNode;
+            }
```

Comment:
We add an explicit `return;` before local functions.

```suggestion
            return;

            static BaseRevisionNode CreatePathNode(Tree tree, string parentPath)
                => new(tree, parentPath);
```